rfd/0155-scoped-webauthn-credentials.md

```diff
@@ -0,0 +1,197 @@
+---
+authors: Brian Joerger (bjoerger@goteleport.com)
+state: draft
+---
+
+# RFD 155 - Scoped Webauthn Credentials
+
+## Required Approvers
+
+- Engineering: @rosstimothy && @codingllama
+- Product: @klizhentas || @xinding33
+- Security: @jentfoo
+
+## What
+
+Add and enforce scope for webauthn credentials.
+
+## Why
+
+Today, when a client performs a webauthn ceremony, the resulting webauthn
+credentials can be used to perform actions other than what the client initially
+intended.
+
+For example, webauthn credentials intended for per-session MFA could be used
+instead for login. This means that if an attacker gets a hold of a user's
+webauthn credentials before the client consumes them, an attacker with the
+user's password could login as the user. Adding scope to the webauthn
+credentials would prevent the attacker from using the stolen webauthn
+credentials freely.
+
+Adding scope also enables us to allow webauthn credentials to be reused within
```

Comment:
Adding to this, we could have clients tell the server when they are done using a challenge, so it can be cleaned up. We may not even need the reuse bool, as some kinds can't be reused and others (like admin challenges) can use the client signal for the early delete.

Reply:
I'm going to omit this from the RFD, but I'll investigate this during implementation as a nice to have.